qmarkdowntextedit.cpp

```diff
@@ -304,6 +304,12 @@ bool QMarkdownTextEdit::eventFilter(QObject *obj, QEvent *event) {
             _searchWidget->doSearch(
                     !keyEvent->modifiers().testFlag(Qt::ShiftModifier));
             return true;
+        } else if ((keyEvent->key() == Qt::Key_Z) &&
+                   (keyEvent->modifiers().testFlag(Qt::ControlModifier)) &&
+                   (keyEvent->modifiers().testFlag(Qt::ShiftModifier))) {
+            //handle this before undo otherwise Ctrl+Shift+Z is caught by Ctrl+Z for some reason
```

Comment:
the reason is because the if-clause simply is true, the `Qt::ControlModifier` is also activated if you are pressing `Ctrl + Shift + Z`

Reply:
Yess...